### ISSUE TITLE
Browser field with additional fields

### ISSUE THREAD

**Issue by zipavlin** [area17/twill#219]
Is it possible to attach additional fields to Browser field?

For example: we have 2 models that are in many-to-many relation: Consultant and Office, where Consultant have a specific role in an Office; like so:

```
Schema::create('consultant_office', function (Blueprint $table) {
    $table->bigIncrements('id');
    $table->timestamps();
    $table->string('role')->default('manager');
    $table->integer('position')->unsigned()->index();
    $table->unsignedInteger('office_id');
    $table->foreign('office_id')->references('id')->on('offices')->onDelete('cascade');
    $table->unsignedInteger('consultant_id');
    $table->foreign('consultant_id')->references('id')->on('consultants')->onDelete('cascade');
});
```

I would like to add a select field for 'role' for each item that is added to browser field. Is it possible to do so? Is there any other way of doing this? I guess I could do it with repeater, but I like functionality that you can only add 1 instance of item in browser.

Thanks for your help!

**Comment by zipavlin** [comment]
I figured this is the same as #164

**Comment by ifox** [comment]
Hi @zipavlin you should be able to use `'max' => 1` when defining repeaters configuration in `config/twill.php`!

**Comment by zipavlin** [comment]
Hi @ifox, wouldn't ```'max' => 1``` just limit repeater field to have only 1 entry?

I would like to reproduce browser field behaviour where user can add multiple entries but they all must be unique.

If I understand correctly, I could do this with repeater by populating a select field inside repeater item with ID's of entities I want to link to in relationship. To limit adding only one entry with specific ID, I would need to remove it from populated select field's options array when that item would be added to repeater.

It's sort of hard to explain...

**Comment by ifox** [comment]
Gotcha. I didn't understand initially that you're looking at uniqueness, not count limit.

We're working on a complex permissions manager internally and have a need for browser with an additional value stored for each row. Similarly to your use case, we attach users and give them a permission level (view, edit, manage). That would most likely work great for you too to select a role on each attached browser rows.

<img width="763" alt="Screen Shot 2019-03-25 at 6 29 35 PM" src="https://user-images.githubusercontent.com/708498/54958607-b415e580-4f2c-11e9-823b-d3bc24bfd1e3.png">

Something like that?

**Comment by zipavlin** [comment]
> a need for browser with an additional value stored for each row

Yes, this exactly! :)

Do you think you could share your component? Or some guides to push me in the right direction?

**Comment by arkid** [comment]
Hi @zipavlin did you get this working?  Anything you can share ?

**Comment by zipavlin** [comment]
@arkid I wrote my own Vue component for this. It doesn't look as nice as Twill and is a bit hacky, but it works for now. I think sharing it will not help you much as it is really tied to that specific requirement.

What I did, basically, is I moved all logic to Vue, send json to server and update DB accordingly (prob. doing another check there, but I am not at work now, so hard to say what I did exactly).
